src/App.tsx

```diff
@@ -1,61 +1,121 @@
+import React, { useState } from 'react';
+import { TodoList } from './components/TodoList/TodoList';
+import { Todo } from './Types/Todo';
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import { getUserById } from './Helpers/getUser';
+import { getNewTodoId } from './Helpers/getNewTodoId';
+
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { User } from './Types/User';
+
+const todosWithUsers: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUserById(todo.userId, usersFromServer),
+}));
+
+export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>(todosWithUsers);
+  const [title, setTitle] = useState<string>('');
+  const [userId, setUserId] = useState<number | string>('');
+  const [isTitleValid, setIsTitleValid] = useState<boolean>(true);
+  const [isUserSelected, setIsUserSelected] = useState<boolean>(true);
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (!title || !userId) {
+      if (!title) {
+        setIsTitleValid(false);
+      }
+
+      if (!userId) {
+        setIsUserSelected(false);
+      }
+
+      return;
+    }
+
+    const newTodo = {
+      id: getNewTodoId(todos),
+      userId: +userId,
+      title,
+      completed: false,
+      user: getUserById(+userId, usersFromServer),
+    };
+
+    setTodos((current => [...current, newTodo]));
```

Comment:
Use more descriptive name.
```suggestion
    setTodos((prevTodos=> [...prevTodos, newTodo]));
```